src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -145,7 +146,7 @@ public SQLiteStore(String directory, Manager manager, StoreDelegate delegate) {
         }
         this.path = new File(directory, kDBFilename).getPath();
         this.manager = manager;
-        this.storageEngine = null;
+        this.storageEngine = createStorageEngine();
```

Comment:
Another way to handle this is checking whether the storage is null or not in derivePBKDF2SHA256Key(String password, byte[] salt, int rounds) method below. If it's null, create a new one. I think this is a better approach. What do you think?